src/Dictionaries/HashedDictionary.cpp

```diff
@@ -465,14 +682,15 @@ void HashedDictionary<dictionary_key_type, sparse>::updateData()
     if (update_field_loaded_block)
```

Comment:
We need to have multiple `update_field_loaded_blocks` or for shared dictionary load this block in multiple threads.

Reply:
I've looked through this updatable dictionaries and the implementation looks odd to me, and I doubt that it is useful for this sharded use cases, so I disabled for now sharded support for sources with `update_field` and added a test.

I will submit another patch about update_field later.